### ISSUE TITLE
Gem status

### ISSUE THREAD

**Issue by hedselu** [rmosolgo/graphql-parallel#3]
Hello, I would like to ask if this gem will be further maintained? Actual version dependency of graphql  is ~> 0.9.4.

**Comment by rmosolgo** [comment]
No, this gem is not maintained, sorry! I tried to capture that in the title of the repo:

![image](https://user-images.githubusercontent.com/2231765/32674118-eb111eac-c61f-11e7-999c-9c4d527cbbfc.png)


But perhaps I should make it more clear!

**Comment by hedselu** [comment]
Thanks for info! You are right, just haven’t noticed this. Some face-punching message like in [here](https://github.com/rmosolgo/graphql-streaming) would be more readable I guess. Btw. why this is no longer maintained? I’ve looked little bit into current GraphQL gem version haven’t seen anything blocking this gem. I mean, I’d like to fork/contribute to this gem if changes are required due to dependencies update and test it in action. But also I’d like to know if abandoning this gem had any significant reason and if any action taken on it will make sense.

**Comment by lokaltrav** [comment]
For anyone else who wants parallelism in graphql queries, read this blog:

http://rmosolgo.github.io/blog/2017/01/22/parallelism-in-graphql-ruby/

It's simple to set up:
add to GemFile:
```
gem 'concurrent-ruby'
```
Add to your root schema definition.
```
GraphSchema = GraphQL::Schema.define do
  query QueryGraph
  mutation MutationGraph
  lazy_resolve(Concurrent::Future, :value)  <--- this is part of the magic
```
That tells GraphQL that when it runs across a result of that class, to call .value on it, which then blocks execution until it's completed.
```
  field :booking_request do
    type BookingRequestGraph

    argument :id, types.ID

    resolve ->(obj, args, ctx) {
      Concurrent::Future.execute { <--- use this around the code you want to execute in a separate thread.
        GraphLoader.load(args[:id], BookingRequest, ctx, :read)
      }
    }
  end
```
Now, when it resolves, it's only returning a Concurrent::Future object which is executing code in a separate thread. The 'resolve' doesn't wait for the code to complete before it returns. So, if you have a large GraphQL query come in where all referenced parts of the graph return Future, then all will be started at nearly the same time and execute in parallel. The query won't return to caller until all Future blocks have finished executing (by waiting for the .value).

Since Concurrent::Future.execute {} is not related to GraphQL at all, you can use it in your code where appropriate to speed things up. I think that also makes it less brittle since it's not tied to any version of GraphQL at all and only slight changes may be needed in the future to the "lazy_resolve" bit so you don't have to worry about a package no longer being maintained and kept in pace with the graphql version you're using.

If it's common for your client code to execute the equivalent of booking request detail records where each of those may cause their own separate queries, you could have huge savings. For my code, I cut the execution time in half by implementing Future blocks since many of the detail records had calls that then looked at their own child relationships and executing them in parallel saved a bunch.

Since there is a cost incurred for starting a separate thread, if you're doing simple one-node queries without anything else able to execute in parallel, it's therefore possible you could slightly slow down your queries, so use only where appropriate.
